### PR TITLE
Noviny/update gatsby source workspaces

### PR DESCRIPTION
**Pull request by Noviny**

### CODE REVIEW

File: packages/gatsby-source-workspace/README.md [hunk 1]
```diff
@@ -2,15 +2,125 @@
...
 
-- the main `README.md`
-- the `CHANGELOG.md`
+```graphql
+type workspaceInfo implements Node {
+  name: String
+  version: String
+  dir: String
+  relativeDir: String
+  packageJSON: JSON!
```
Comment (sarahbethfederman) [inline packages/gatsby-source-workspace/README.md]: ❤️

File: test-gatsby/gatsby-config.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
 const findWorkspacesRoot = require("find-workspaces-root").default;
+const fs = require("fs-extra");
+const path = require("path");
 
 // Funfact, all gatsby code and example code uses template literlas instead of
 // strings - see if you can figure out the places I copy/pasted code
```
Comment (sarahbethfederman) [inline test-gatsby/gatsby-config.js]: Does this comment serve a purpose? 🙃

File: packages/gatsby-source-workspace/README.md [hunk 3]
```diff
@@ -2,15 +2,125 @@
...
+
+module.exports = {
+  plugins: [
+    {
+      resolve: "@manypkg/gatsby-source-workspace",
+      options: {
+        extraFields: [
+          {
+            name: "readme",
+            definition: `String`,
```
[the remark this answers is not in the document]
Comment (sarahbethfederman) [reply packages/gatsby-source-workspace/README.md]: nvm I answered my own question https://www.gatsbyjs.org/docs/actions/#createTypes

**Comment by Noviny** [comment]
> Do we need commands for test-gatsby added in a package somewhere?

There's a script to start it in `develop` mode, but maybe we need more things?

It's worth calling out that this package's construction has quite a bit of fast + loose about it, such as:

- No tests
- No types
- very little previous code review

I haven't chose to address these things here, or look at other repo-level concerns. It's possible this is important enough that we want it as a ticket?

File: test-gatsby/gatsby-config.js (hunk 2 above)
Comment (Noviny) [reply test-gatsby/gatsby-config.js]: Entertainment value?

**Pull request merged**